### ISSUE TITLE
Show Impls instead of all functions

### ISSUE THREAD

**Issue by douweschulte** [Enselic/public-api#83]
As part of the tree data structure I was thinking about showing the `impl`s of a `struct`/`enum`/`trait` instead of the full list of functions. For local traits the functions will already be in the list (in the trait definition) and for other traits the functions can be assumed to be known or at least less part of this crates api and more for that other crate. Also this should remove clutter if only a `trait` is updated this should only list this change one time and not for every entity implementing it.

Implementing this change would be a nice first step for a more tree like data structure as discussed before. I have already tried some things locally but nothing is working yet.

Mock up:
```rust
#[derive(Clone, Copy, Debug)]
pub struct Plain {
    pub x: usize,
}
```
Output:
```
pub struct comprehensive_api::structs::Plain
Implements: Clone, Copy, Debug
```

**Comment by Enselic** [comment]
I completely agree that it very well could make sense to show what traits are implemented rather than what functions (of traits) that are implemented. I'm not sure either exactly how to present it or diff it. But please continue experimenting and let me know if you figure out a good way :)
